collector/src/test/java/com/google/cloud/trace/zipkin/ZipkinCollectorIntegrationTest.java

```diff
@@ -0,0 +1,102 @@
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+import org.springframework.core.task.SyncTaskExecutor;
+import org.springframework.test.context.junit4.SpringRunner;
+import zipkin.Codec;
+import zipkin.Span;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.Executor;
+
+import static com.google.cloud.trace.zipkin.StackdriverZipkinCollector.ZIPKIN_CONFIG_NAMES;
+import static java.lang.Math.min;
+import static java.util.Arrays.asList;
+import static java.util.Collections.unmodifiableList;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.hasSize;
+import static org.mockito.Mockito.mock;
+import static zipkin.TestObjects.LOTS_OF_SPANS;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = {StackdriverZipkinCollector.class, ZipkinCollectorIntegrationTest.TestConfiguration.class},
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+    properties = {ZIPKIN_CONFIG_NAMES, "zipkin.storage.stackdriver.project-id=test_project"})
+public class ZipkinCollectorIntegrationTest
+{
+  //LOTS_OF_SPANS is too much! I need at most 30!
+  private static final List<Span> TEST_SPANS = unmodifiableList(asList(Arrays.copyOfRange(LOTS_OF_SPANS, 0, min(30, LOTS_OF_SPANS.length))));
+
+  @ClassRule
+  public static final StackdriverTraceClassRule traceService = new StackdriverTraceClassRule();
+
+  @ClassRule
+  public static final ProvideSystemProperty hostProperty
+      = new ProvideSystemProperty("zipkin.storage.stackdriver.api-host", traceService.grpcURI());
+
+  @Autowired
+  private TestRestTemplate restTemplate;
+
+  @Test
+  public void traceSinkGetsCalled() throws InterruptedException
+  {
+    final Set<Long> spanIds = toIds(TEST_SPANS);
+    assertThat("Expected to test at least one span", spanIds, hasSize(greaterThan(0)));
+    assertThat("Unexpected trace in Stackdriver", traceService.spanIds(), Matchers.<Long>empty());
+
+    for (List<Span> partition : Lists.partition(TEST_SPANS, 3))
+    {
+      this.restTemplate.postForLocation("/api/v1/spans", Codec.JSON.writeSpans(partition));
+    }
+
+    assertThat("Not all spans made it to Stackdriver", spanIds, equalTo(traceService.spanIds()));
+  }
+
+  private Set<Long> toIds(List<Span> spans)
+  {
+    final Set<Long> spanIds = new HashSet<>();
+    for (Span span : spans)
+    {
+      spanIds.add(span.id);
+    }
+    return spanIds;
+  }
+
+  @Configuration
+  public static class TestConfiguration
+  {
+    @Bean(name = "stackdriverExecutor")
+    @Primary
+    public Executor syncExecutor()
+    {
+      //executes each task synchronously
+      return new SyncTaskExecutor();
```

Comment:
nice work, so one last thing you can consider is using a blocking queue or otherwise in the class rule. That way, you can run tests with all defaults except auth. (ex you won't need to rely on a synchronous executor, as a blocking call to takePatchRequest() will block until the thing happened).